core/src/lib.rs

```diff
@@ -1,4 +1,6 @@
 //! Iroha — A simple, enterprise-grade decentralized ledger.
+
+#![allow(incomplete_features)] // This feature is minor and breakage to be expected is going to be minimal.
 #![feature(adt_const_params)]
```

Comment:
additional possible reason is that with `nightly` compiler I have a feeling that build cache is cleared much more often. I am not 100% sure of this, since it's still rather fast on M1

Reply:
M1 is faster than it has any right to be. We shouldn't expect people to have it though.